kv/replica_slice_test.go

```diff
@@ -110,3 +111,27 @@ func TestReplicaSetMoveToFront(t *testing.T) {
 		t.Errorf("expected order %s, got %s", exp, stores)
 	}
 }
+
+func TestReplicaSetRandomOrder(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	rs := replicaSlice(nil)
+	for i := 0; i < 10; i++ {
+		rs = append(rs, replicaInfo{ReplicaDescriptor: roachpb.ReplicaDescriptor{StoreID: roachpb.StoreID(i + 1)}})
+	}
+
+	rs.SortRandomOrder(2, 10)
```

Comment:
I figured since TestReplicaSetSortByCommonAttributePrefix is already testing SortByCommonAttributePrefix, and SortRandomOrder is inside, it should already check the ordering with random order function has called. While I was testing with SortRandomOrder function, I have seen some error messages related of expected ordering when shuffle didn't go well. 

Also I have tried calling TestReplicaSetSortByCommonAttributePrefix fund inside TestReplicaSetMoveToFront just to see if anything goes wrong, but it seems duplicated to call the verifyOrdering though. Or did you mean something else?